File: python_cpp/uPIMulator_frontend/src/abi/directive/long_directive.py
from abi.word.immediate import Immediate
from abi.word.representation import Representation


class LongDirective:
    def __init__(self, value: int):
        self._value: Immediate = Immediate(Representation.UNSIGNED, 8 * LongDirective.size(), value)

    def value(self) -> int:
        return self._value.value()

    @staticmethod
    def size() -> int:
        return 4

File: golang/uPIMulator/src/linker/kernel/directive/long_directive.go
package directive

import (
	"uPIMulator/src/abi/encoding"
	"uPIMulator/src/abi/word"
)

type LongDirective struct {
	immediate *word.Immediate
}

func (this *LongDirective) Init(value int64) {
	this.immediate = new(word.Immediate)
	this.immediate.Init(word.UNSIGNED, 8*int(this.Size()), value)
}

func (this *LongDirective) Size() int64 {
	return 4
}

func (this *LongDirective) Immediate() *word.Immediate {
	return this.immediate
}

func (this *LongDirective) Encode() *encoding.ByteStream {
	return this.immediate.ToByteStream()
}
